azurerm/resource_arm_service_fabric_cluster.go

```diff
@@ -509,6 +543,49 @@ func expandServiceFabricClusterAddOnFeatures(input []interface{}) *[]string {
 	return &output
 }
 
+func expandServiceFabricClusterAzureActiveDirectory(input []interface{}) *servicefabric.AzureActiveDirectory {
+	if len(input) == 0 {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+
+	tenantId := v["tenant_id"].(string)
+	clusterApplication := v["cluster_application_id"].(string)
+	clientApplication := v["client_application_id"].(string)
+
+	config := servicefabric.AzureActiveDirectory{
+		TenantID:           utils.String(tenantId),
+		ClusterApplication: utils.String(clusterApplication),
+		ClientApplication:  utils.String(clientApplication),
+	}
+	return &config
+}
+
+func flattenServiceFabricClusterAzureActiveDirectory(input *servicefabric.AzureActiveDirectory) []interface{} {
+	results := make([]interface{}, 0)
+
+	if v := input; v != nil {
```

Comment:
_minor_ we tend to invert these e.g.

```
if input == nil {
  return []interface{}{}
}

output := make(map[string]interface{}, 0)
if v := input.TenantID; v != nil {
  output["tenant_id"] = *v
}
if v := input.ClusterApplication; v != nil {
  output["cluster_application_id"] = *v
}
if v := input.ClientApplication; v != nil {
  output["client_application_id"] = *v
}

return []interface{}{output}
```

Reply:
@tombuildsstuff . With out refactoring the whole file i'd say that current version better fits with the other functions, making it an easier read from start to end. But i'm happy to make the change if you think its still best todo so.

Reply:
it's not a blocker at all, more just for consistency with the other resources, but logically it's the same :)